hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -189,6 +188,14 @@ instance HasKnownUTxO ChainState where
     Open st -> getKnownUTxO st
     Closed st -> getKnownUTxO st
 
+-- | Defines the starting state of the direct chain layer.
+initialChainState :: ChainStateType Tx
+initialChainState =
+  ChainStateAt
+    { chainState = Idle
+    , recordedAt = Nothing
+    }
```

Comment:
As we are refactoring the chain state handling here, I think we should get rid of in the same go.

The `Idle` chain state does not contain any information. Neither does such a chain state exist for other chain instances (`SimpleTx`) and we can get rid of this completely by making the `ChainStateHistory` a normal list and starting it with `[]`.

Reply:
Idle actually records information that we are in Idle just after Abort and Fanout.

This is in fact the reason why we can't just say we should never be on a Maybe ChainState.

Reply:
Leaving the refactor of making the `ChainStateHistory` a normal list and starting it with `[]` for another PR.